src/routes/channels/[id]/+page.svelte

```diff
@@ -0,0 +1,491 @@
+<script lang="ts">
+  import { fade, slide } from 'svelte/transition'
+  import { channels$, lastPath$, nodeInfo$, onDestroy$, settings$ } from '$lib/streams'
+  import { goto } from '$app/navigation'
+  import BackButton from '$lib/elements/BackButton.svelte'
+  import { translate } from '$lib/i18n/translations'
+  import type { PageData } from './$types'
+  import CopyValue from '$lib/elements/CopyValue.svelte'
+  import SummaryRow from '$lib/elements/SummaryRow.svelte'
+  import Spinner from '$lib/elements/Spinner.svelte'
+  import { channelStatusTolabel } from '../utils.js'
+  import { BitcoinDenomination, type Channel } from '$lib/types.js'
+  import { formatValueForDisplay } from '$lib/utils.js'
+  import { convertValue } from '$lib/conversion.js'
+  import { currencySymbols } from '$lib/constants.js'
+  import Big from 'big.js'
+  import Button from '$lib/elements/Button.svelte'
+  import Modal from '$lib/elements/Modal.svelte'
+  import TextInput from '$lib/elements/TextInput.svelte'
+  import edit from '$lib/icons/edit.js'
+  import lightning from '$lib/lightning.js'
+  import ErrorMsg from '$lib/elements/ErrorMsg.svelte'
+  import { filter, map, switchMap, takeUntil, timer } from 'rxjs'
+  import connect from '$lib/icons/connect.js'
+
+  export let data: PageData // channel id
+
+  let channel: Channel
+
+  let feeBaseUpdate: number
+  let feeRateUpdate: number
+  let minForwardUpdate: number
+  let maxForwardUpdate: number
+
+  $: if ($channels$.data) {
+    findChannel()
+  }
+
+  function findChannel() {
+    channel = $channels$.data!.find((p) => p.id === data.id)!
+
+    if (channel) {
+      const { feeBase, feePpm, htlcMin, htlcMax } = channel
+
+      if (feeBase) {
+        feeBaseUpdate = Big(feeBase).div(1000).toNumber()
+      }
+
+      if (feePpm) {
+        feeRateUpdate = feePpm
+      }
+
+      if (htlcMin) {
+        minForwardUpdate = Big(htlcMin).div(1000).toNumber()
+      }
+
+      if (htlcMax) {
+        maxForwardUpdate = Big(htlcMax).div(1000).toNumber()
+      }
+    }
+  }
+
+  const lnApi = lightning.getLn()
+
+  timer(5000, 10000)
+    .pipe(
+      filter(() => !!channel),
+      switchMap(() => lnApi.getChannels(channel.peerId, channel.id)),
+      map(([update]) => update),
+      takeUntil(onDestroy$)
+    )
+    .subscribe((channelUpdate) => {
+      channel = channelUpdate
+    })
```

Comment:
Yep that is what the `takeUntil` operator does. It will automatically unsubscribe when the component is destroyed. But  yeah that is always my first thought when I see a subscribe as well.